### ISSUE TITLE
[ENHANCEMENT] Load balancer support http(s) health checks

### ISSUE THREAD

**Issue by johnwc** [yasker/vm#6100]
**Is your enhancement related to a problem? Please describe.**
We had an SSL certificate issue on one of our VMs that was participating in a harvester load balancer. The load balancer only checks for port connectivity for health checks, so it was continuing to forward requests to the problematic VM.

**Describe the solution you'd like**
Have the health checks for harvester VM load balancer also support checking via http(s) request.

**Describe alternatives you've considered**
N/A

**Additional context**
Would also be great to have a tab in the LB details page showing which VM is active or not active.

**Comment by w13915984028** [comment]
@johnwc It looks a bit complex to probe https regarding SSL/cerficated related areas.

Could you add more details about your SSL issue? thanks.

Harvester LB supports TCP connection based health check, this does not rely on others.

**Comment by johnwc** [comment]
@w13915984028  We had one of our VMs fail to update the ACME SSL for the site on the VM, so the cert expired. Causing the load balancer to randomly serve up a site with an invalid SSL certificate to our client's browser.

Could you not use a library like [this one](https://github.com/lifeforms/httpcheck?tab=readme-ov-file#library-example) when the healthcheck is a url? 
Or utilize the [same one](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober.go) that is used within K8s instead of [tevino/tcp-shaker](https://github.com/tevino/tcp-shaker)?

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by johnwc** [comment]
Feature request

**Comment by ihcsim** [comment]
@johnwc Will it work if we provide a way to define [liveness and readiness probes](https://kubevirt.io/user-guide/user_workloads/liveness_and_readiness_probes/#define-readiness-probes) for the vm? I don't see why we need to implement these probes ourselves when K8s already provided them, and KubeVirt supports it.

With [readiness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes) defined, if an endpoint returns anything greater than HTTP 400, K8s will temporary stop any inbound to the associated `Service`.

**Comment by johnwc** [comment]
Will the health probe fail if it connects to a HTTPS with invalid certificate? I didn't think that the load balancer in harvester was using KubeVirt?

**Comment by ihcsim** [comment]
Yes, the probe let you define [arbitrary shell commands](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command) to determine the service health, as long as the tools used by the command are installed in your VM. Assuming the cert is on the vm disk, you can use `openssl` to determine its expiry date and use some bash-fu to determine if it has expired. Or you can configure the probe to `curl` the endpoint if you know the IP or DNS name of the VM ahead of time. I have an issue at https://github.com/harvester/harvester/issues/9523 to spike out some of these use cases.

**Comment by johnwc** [comment]
But the load balancer in harvester does not use KubeVirt.

**Comment by ihcsim** [comment]
The LB is just a L4 K8s `Service` that proxies inbound to your virtual machines. KubeVirt is never involved there. If your on-disk TLS cert is expired, the readiness probe marks your virtual machine is not ready. Then K8s automatically stops proxying inbound to that virtual machine.

**Comment by johnwc** [comment]
Not stale

**Comment by johnwc** [comment]
Not stale

**Comment by johnwc** [comment]
I give up
